extensions/twingate/src/api/twingate.ts

```diff
@@ -0,0 +1,365 @@
+import { GraphQLClient } from "graphql-request";
+import { getPreferenceValues } from "@raycast/api";
+import { ResourceListItem } from "../types.js";
+import { DebugLogger } from "../utils/debug";
+
+// Enhanced error types for better error handling
+export enum TwingateErrorType {
+  AUTHENTICATION = "authentication",
+  NETWORK = "network",
+  API_LIMIT = "api_limit",
+  TIMEOUT = "timeout",
+  INVALID_RESPONSE = "invalid_response",
+  UNKNOWN = "unknown",
+}
+
+export class TwingateError extends Error {
+  constructor(
+    public type: TwingateErrorType,
+    public message: string,
+    public originalError?: Error,
+  ) {
+    super(message);
+    this.name = "TwingateError";
+  }
+}
+
+interface Preferences {
+  apiKey: string;
+  networkUrl: string;
+}
```

Comment:
You don't need to manually set preference types as this is autogenerated in `raycast-env.d.ts`